valik kod/lab2.cpp

```diff
@@ -0,0 +1,161 @@
+#include <iostream>
+#include <initializer_list>
+#include <functional>
+#include <memory>
+#include <stdexcept>
+
+template <typename T>
+class LinkedList
+{
+private:
+    struct Node
+    {
+        T data;
+        std::unique_ptr<Node> next;
+        Node(const T &data) : data(data), next(nullptr) {}
+    };
+
+    std::unique_ptr<Node> head;
+    size_t size_;
+
+public:
+    LinkedList() : head(nullptr), size_(0) {}
+
+    LinkedList(std::initializer_list<T> list) : LinkedList()
+    {
+        auto it = list.end();
+        while (it != list.begin())
+        {
+            --it;
+            push_front(*it);
+        }
+    }
+
+    LinkedList(LinkedList &&other) noexcept : head(std::move(other.head)), size_(other.size_)
+    {
+        other.size_ = 0;
+    }
+
```

Comment:
Method Definitions: Placed the opening brace { of method definitions on the same line as the method declaration.

Reply:
ok